### PR TITLE
topology: nhlt: ssp-nhlt: Set link clock source based on bclk_freq

### PR DESCRIPTION
**Pull request by ranj063**
There is no need to set this in topology as this can be derived based on whether the bit clock frequency is derived from 19.2MHz or 24.576MHz.

### CODE REVIEW

File: topology/nhlt/intel/ssp-nhlt.c [hunk 1]
```diff
@@ -283,14 +266,32 @@ static int set_aux_params(struct intel_nhlt_params *nhlt, snd_config_t *cfg, snd
...
+		return ret;
+
+	/*
+	 * if the bit clock frequency is derived from 19.2MHz, set clock source to 0
+	 * or if the bit clock frequency is derived from 24.576MHz, set clock source to 1
+	 */
+	if (!(19200000 % bclk_freq))
+		ssp_link_set_params(nhlt, 0);
+	else if (!(24576000 % bclk_freq))
+		ssp_link_set_params(nhlt, 1);
```
Comment (kv2019i) [inline topology/nhlt/intel/ssp-nhlt.c:294]: Not 100% sure we want to hardcode this in the plugin (but agree this is a cleaner implementation in topology).

Comment (lgirdwood) [reply topology/nhlt/intel/ssp-nhlt.c:294]: yeah, we should not hardcode HW specific details as this assumes the BCLK never has a fractional divider from MCLK.

Comment (plbossart) [reply topology/nhlt/intel/ssp-nhlt.c:294]: it never does @lgirdwood, we never used M/N dividers. Integer division is the only thing we've ever used.

**Comment by plbossart** [comment]
> There is no need to set this in topology as this can be derived based on whether the bit clock frequency is derived from 19.2MHz or 24.576MHz.

@ranj063 this is a valid assumption if the SOC provides the MCLK, likely 99% of usages.

However, there's also the possibility of using an external clock, see "011b: MCLK input" so we need a escape to deal with this case as well.

Sorry to bring the exception to the rule, I just thought about it now. 

If the goal is to avoid setting this all the time, I wonder if we should have a default in the topology base class and override it as needed. It might be more future proof that a change of tools required?

**Comment by ranj063** [comment]
closing this one in favor of https://github.com/thesofproject/sof/pull/7370. Thanks everyone for the feedback

**Pull request closed without merging**